File: spring/prog_terminal_marittimo/src/main/java/progetto_terminal_marittimo/viaggiRegistrati.java
package progetto_terminal_marittimo;

import java.sql.Date;

public class viaggiRegistrati {
    private String nomeNave;
    private Date data_partenza;
    private Date data_arrivo;
    private int porto_partenza;
    private int porto_arrivo;
    private String direttrice;
    private int addetto;


    public viaggiRegistrati(Date dp, Date da, int pp, int pa, String d, int add, String n){
        this.nomeNave=n;
        this.data_partenza=dp;
        this.data_arrivo=da;
        this.porto_partenza=pp;
        this.porto_arrivo=pa;
        this.direttrice=d;
        this.addetto=add;
    }

    public String getNome(){
        return this.nomeNave;
    }
    
    public Date getDataPartenza(){
        return this.data_partenza;
    }
    
    public Date getDataArrivo(){
        return this.data_arrivo;
    }
    
    public int getPortoPartenza(){
        return this.porto_partenza;
    }
    
    public int getPortoArrivo(){
        return this.porto_arrivo;
    }
    
    public String getDirettrice(){
        return this.direttrice;
    }
    
    public int getAddetto(){
        return this.addetto;
    }
}

File: pagine/js/viaggiRegistrati.js
class ViaggiRegistrati {
    constructor(dataPartenza, dataArrivo, portoPartenza, portoArrivo, direttrice, addetto, nomeNave) {
        this.nomeNave = nomeNave;
        this.data_partenza = new Date(dataPartenza);
        this.data_arrivo = new Date(dataArrivo);
        this.porto_partenza = portoPartenza;
        this.porto_arrivo = portoArrivo;
        this.direttrice = direttrice;
        this.addetto = addetto;
    }

    getNome() {
        return this.nomeNave;
    }

    getDataPartenza() {
        return this.data_partenza;
    }

    getDataArrivo() {
        return this.data_arrivo;
    }

    getPortoPartenza() {
        return this.porto_partenza;
    }

    getPortoArrivo() {
        return this.porto_arrivo;
    }

    getDirettrice() {
        return this.direttrice;
    }

    getAddetto() {
        return this.addetto;
    }

    toString() {
        return "Viaggio con nave: " + this.nomeNave +", partenza: " + this.data_partenza.toISOString().split("T")[0] +", arrivo: " + this.data_arrivo.toISOString().split("T")[0] +", porto partenza: " + this.porto_partenza +", porto arrivo: " + this.porto_arrivo +", direttrice: " + this.direttrice +", addetto: " + this.addetto;
    }
}



